server/service/appconfig.go

```diff
@@ -134,13 +134,32 @@ func getAppConfigEndpoint(ctx context.Context, request interface{}, svc fleet.Se
 		return nil, err
 	}
 
-	// Only the Global Admin should be able to see see SMTP, SSO and osquery agent settings.
+	isGlobalAdmin := vc.User.GlobalRole != nil && *vc.User.GlobalRole == fleet.RoleAdmin
+	isTeamAdmin := false
+	if vc.User.Teams != nil {
+		// gather the teams the current user is admin for.
+		currentUserTeamAdmin := make(map[uint]struct{})
```

Comment:
AFAICS there's no need to assemble a map, just use a boolean `isAdminOfATeam := false`.